server/src/main/java/org/elasticsearch/index/query/TermQueryBuilder.java

```diff
@@ -129,6 +129,19 @@ public static TermQueryBuilder fromXContent(XContentParser parser) throws IOExce
         }
         return termQuery;
     }
+    
+    @Override
+    protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws IOException {
+        if ("_index".equals(fieldName)) {
+            // Special-case optimisation for canMatch phase:  
+            // We can skip querying this shard if the index name doesn't match the value of this query on the "_index" field.
+            QueryShardContext shardContext = queryRewriteContext.convertToShardContext();
+            if (shardContext != null && shardContext.indexMatches(BytesRefs.toString(value)) == false) {
```

Comment:
`BytesRefs.toString` is not needed ?
```suggestion
            if (shardContext != null && shardContext.indexMatches(value) == false) {
```

Reply:
Unlike wildcardquery `value` is an object not a string. I remember having a test failure where I assumed String and it was a BytesRef so had to add this

Reply:
The prefix change suggestion looks to have fixed things, thanks